contrib/epee/include/net/http_protocol_handler.inl

```diff
@@ -612,6 +616,18 @@ namespace net_utils
 				m_want_close = true;
 			}
 		}
+
+		// Cross-origin resource sharing
+		if(m_query_info.m_header_info.m_origin.size())
+		{
+			auto it = std::find(m_config.m_access_control_origins.begin(), m_config.m_access_control_origins.end(), m_query_info.m_header_info.m_origin);
```

Comment:
> If the value of the Origin header is not a case-sensitive match for any of the values in list of origins, do not set any additional headers and terminate this set of steps.

From https://www.w3.org/TR/cors/#resource-requests so I think it's supposed to be case sensitive. 

> Two origins are "the same" if, and only if, they are identical.  In
   particular:
> -  If the two origins are scheme/host/port triples, the two origins
      are the same if, and only if, they have identical schemes, hosts,
      and ports.
> -  An origin that is a globally unique identifier cannot be the same
      as an origin that is a scheme/host/port triple.

https://tools.ietf.org/html/rfc6454#section-4

Making all of the requested changes at the moment. I'll look into making callable structs for this as well. Going to keep it case sensitive for now based on that above doc although I agree it's odd personally that it isn't case insensitive.

Reply:
Seems unexpected since domains are case insensitive. However, its probably easier for unicode purposes, and clients that send upper case characters are likely rare.